zone/doors.cpp

```diff
@@ -685,81 +687,23 @@ void Doors::ToggleState(Mob *sender)
 	safe_delete(outapp);
 }
 
-int32 ZoneDatabase::GetDoorsCount(uint32 *oMaxID, const char *zone_name, int16 version)
+uint32 ZoneDatabase::GetDoorsCountPlusOne()
 {
+	const uint32 max_door_id = static_cast<uint32>(DoorsRepository::GetMaxId(*this));
 
-	std::string query   = StringFormat(
-		"SELECT MAX(id), count(*) FROM doors "
-		"WHERE zone = '%s' AND (version = %u OR version = -1)",
-		zone_name, version
-	);
-	auto        results = QueryDatabase(query);
-	if (!results.Success()) {
-		return -1;
-	}
-
-	if (results.RowCount() != 1) {
-		return -1;
-	}
-
-	auto row = results.begin();
-
-	if (!oMaxID) {
-		return atoi(row[1]);
-	}
-
-	if (row[0]) {
-		*oMaxID = atoi(row[0]);
-	}
-	else {
-		*oMaxID = 0;
-	}
-
-	return atoi(row[1]);
-
+	return (max_door_id + 1);
 }
 
-int32 ZoneDatabase::GetDoorsCountPlusOne(const char *zone_name, int16 version)
+int ZoneDatabase::GetDoorsDBCountPlusOne(std::string zone_short_name, int16 version)
 {
-	std::string query   = StringFormat(
-		"SELECT MAX(id) FROM doors WHERE zone = '%s' AND version = %u",
-		zone_name,
+	const auto query = fmt::format(
+		"SELECT COALESCE(MAX(doorid), 0) FROM doors "
```

Comment:
Wouldn't you want this value to be at least `1` instead of `0`

Reply:
True enough.